src/language-handlebars/utils.js

```diff
@@ -1,16 +1,80 @@
 "use strict";
 
+function isUppercase(string) {
+  return string.toUpperCase() === string;
+}
+
 function isGlimmerComponent(node) {
-  if (node.type !== "ElementNode") {
-    return false;
-  }
+  return (
+    isNodeOfSomeType(node, ["TextNode"]) &&
+    typeof node.tag === "string" &&
+    (isUppercase(node.tag[0]) || node.tag.includes("."))
+  );
+}
+
+function isWhitespaceNode(node) {
+  return isNodeOfSomeType(node, ["TextNode"]) && !/\S/.test(node.chars);
+}
+
+function isNodeOfSomeType(node, types) {
+  return node && types.some(type => node.type === type);
+}
+
+function isParentOfSomeType(path, types) {
+  const parentNode = path.getParentNode(0);
+  return isNodeOfSomeType(parentNode, types);
+}
 
-  const tagFirstChar = node.tag && node.tag[0];
-  const isLocal = node.tag.includes(".");
+function isPreviousNodeOfSomeType(path, types) {
+  const previousNode = getPreviousNode(path);
+  return isNodeOfSomeType(previousNode, types);
+}
+
+function isNextNodeOfSomeType(path, types) {
+  const nextNode = getNextNode(path);
+  return isNodeOfSomeType(nextNode, types);
+}
+
+function getPreviousNode(path, lookBack = 1) {
+  const node = path.getValue();
+  const parentNode = path.getParentNode(0) || {};
+  const children = parentNode.children || parentNode.body || [];
+  const index = children.indexOf(node);
+  return index !== -1 && children[index - lookBack];
+}
+
+function getNextNode(path) {
+  const node = path.getValue();
+  const parentNode = path.getParentNode(0);
+  const children = parentNode.children || parentNode.body || [];
+  const index = children.indexOf(node);
+  return index !== -1 && children[index + 1];
+}
+
```

Comment:
Looks like my review get lost, some how,

Make a `getSiblingNode(path, offset)`, and `getNextNode`, `getPreviousNode` as shortcuts

Reply:
CI was red because of a typo

Good advice, I'll add it.

Reply:
Done, neat refactoring in the end. Thanks for your help :)